pkg/kubelet/volumemanager/reconciler/reconciler_common.go

```diff
@@ -277,10 +288,17 @@ func (rc *reconciler) waitForVolumeAttach(volumeToMount cache.VolumeToMount) {
 }
 
 func (rc *reconciler) unmountDetachDevices() {
+	attachedVolumes := []cache.AttachedVolume{}
 	for _, attachedVolume := range rc.actualStateOfWorld.GetUnmountedVolumes() {
+		if rc.operationExecutor.IsOperationPending(attachedVolume.VolumeName, nestedpendingoperations.EmptyUniquePodName, nestedpendingoperations.EmptyNodeName) {
+			continue
+		}
+		attachedVolumes = append(attachedVolumes, attachedVolume)
+	}
+
+	for _, attachedVolume := range attachedVolumes {
```

Comment:
Also this refactoring is distracting. I'd suggest to not do this or at least put all refactoring with no functional changes into a separate commit